build/helper/metadata_add_all.py

```diff
@@ -109,36 +109,57 @@ def _add_is_error_handling(f):
     return f
 
 
-def _add_buffer_info(parameter):
-    '''Adds buffer information to the parameter metadata iff 'size' is defined else assume not a buffer'''
-
-    # For simplicity, we are going to treat ViChar[], ViString, ViConstString, and ViRsrc the same: As ViChar
-    # and is_buffer True
-    t = parameter['type']
-    if (t.find('[ ]') > 0) or (t.find('[]') > 0):
-        assert 'is_buffer' not in parameter or parameter['is_buffer'] is True, 'Conflicting metadata - [] found but is_buffer already set to False.'
-        parameter['type'] = t.replace('[ ]', '').replace('[]', '')
-        parameter['original_type'] = t
-        parameter['is_buffer'] = True
+def _add_buffer_info(parameter, config):
+    '''Adds buffer information to the parameter metadata
+
+    There are 4 different 'is_*' variables we will set
+        'is_string' - Used for any string type - see below for complete list
+        'is_list'   - Used for an array of custom types. We are not putting custom types into array.array or numpy.array
+        'is_array'  - Used for arrays of simple types
+        'is_buffer' - True when either 'is_list' or 'is_array' is True
+    '''
+
+    assert 'is_buffer' not in parameter or not parameter['is_buffer'], "if 'is_buffer' is in metadata then it must be set to False"
+    assert 'is_string' not in parameter, "'is_string' should not be set by metadata or in addons"
+    assert 'is_array' not in parameter, "'is_array' should not be set by metadata or in addons"
+    assert 'is_list' not in parameter, "'is_list' should not be set by metadata or in addons"
+
+    is_string = False
+    is_array = False
+    is_list = False
+    ot = parameter['type']
```

Comment:
what is ot?

Reply:
I see... original type. Might as well spell it out.

Reply:
Done